Sources/WASI/WASI.swift

```diff
@@ -1,4 +1,12 @@
-import Foundation
+#if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
+import Darwin
+#elseif os(Linux) || os(FreeBSD) || os(Android)
+import Glibc
+#elseif os(Windows)
+import ucrt
+#else
+#error("Unsupported Platform")
+#endif
```

Comment:
How come we need to use libc here? Seems like the only use site is `open(..., O_DIRECTORY)` — can we not use `SystemPackage.FileDescriptor.OpenOptions.directory` instead?

Reply:
I wanted to keep using `memchr` because llvm recognizes the lib call and optimizes it.

Reply:
I see, what do you think about making a `memchr` shim in SystemExtras? Might be nice to isolate the C library stuff to that one spot. We can do the same for `O_DIRECTORY`, which seems like it might need to be re-declared by us on Windows (other platforms define `.directory` in SystemPackage).

Reply:
Yes, importing it through a C shim could be an option. Actually `memchr` is not a syscall and `SystemExtras` will be upstreamed to swift-system later, so `SystemExtras` might not be an appropriate place.
Once we upgrade the minimum supported Swift version to 6.0, we can import it via `@_extern(c)` without C module, but it's not feasible right now.

Reply:
Ah interesting, I would've assumed that `UnsafeRawBufferPointer.firstIndex(of:)` could be optimized to `memchr` under the hood but it looks like that isn't being done yet. It's being tracked via https://github.com/apple/swift/issues/63200.

Reply:
Yeah, it's unfortunate situation now 🥲